File: src/main/kotlin/Java/Rol/Mago.java
package Java.Rol;

class Mago extends Personaje {
    private Hechizo hechizo;

    public Mago(String nombre) {
        super(nombre, 100, 10);
        this.hechizo = new Hechizo("Fuego", 25);
    }

    @Override
    public void atacar(Personaje objetivo) {
        System.out.println(nombre + " lanza un hechizo de " + hechizo.getNombre() + " a " + objetivo.getNombre() + ".");
        objetivo.recibirDanio(hechizo.getPoder());
    }

    public void usarHechizo(Personaje objetivo) {
        System.out.println(nombre + " utiliza el hechizo " + hechizo.getNombre() + " contra " + objetivo.getNombre() + ".");
        objetivo.recibirDanio(hechizo.getPoder());
    }
}

File: src/main/kotlin/kotlinOPP/Rol/Mago.kt
package kotlinOPP.Rol

class Mago(nombre: String) : Personaje(nombre, 80) {
    private val hechizos = listOf(
        Hechizo("Bola de Fuego", 15),
        Hechizo("Rayo", 25)
    )

    override fun atacar(): Int {
        val hechizo = hechizos.random()
        println("$nombre lanza ${hechizo.nombre} y causa ${hechizo.danio} de daño.")
        return hechizo.danio
    }

    override fun recibirDanio(danio: Int) {
        puntosDeVida -= danio
        println("$nombre recibe $danio de daño. Puntos de vida restantes: $puntosDeVida")
    }
}